src/pudl/output/ferc1.py

```diff
@@ -1912,12 +1919,71 @@ def forest(self: Self) -> nx.DiGraph:
         if pruned_nodes := set(nodes_before_pruning).difference(nodes_after_pruning):
             raise AssertionError(f"Unexpectedly pruned stepchildren: {pruned_nodes=}")
 
-        # forest = self.set_forest_attributes(
-        #    forest,
-        #    exploded_meta=self.exploded_meta,
-        #    exploded_calcs=self.exploded_calcs,
-        #    tags=self.tags,
-        # )
+        # HACK alter.
+        # two different parents. those parents have different sets of dimensions.
+        # sharing some but not all of their children so they weren't caught from in the
+        # only stepchildren node removal from above. a generalization here would be good
+        almost_pure_stepparents = [
+            NodeId(
+                "utility_plant_summary_ferc1",
+                "depreciation_amortization_and_depletion_utility_plant_leased_to_others",
+                "total",
+                pd.NA,
+                pd.NA,
+            ),
+            NodeId(
+                "utility_plant_summary_ferc1",
+                "depreciation_and_amortization_utility_plant_held_for_future_use",
+                "total",
+                pd.NA,
+                pd.NA,
+            ),
+            NodeId(
+                "utility_plant_summary_ferc1",
+                "utility_plant_in_service_classified_and_unclassified",
+                "total",
+                pd.NA,
+                pd.NA,
+            ),
+        ]
+        forest.remove_nodes_from(almost_pure_stepparents)
+
+        # Ensure that we haven't removed any calculation components that would have
+        # altered the final root-to-leaf calculations:
+        assert (
+            self.exploded_calcs.set_index(self.calc_cols)
+            .loc[pure_stepparents + almost_pure_stepparents, "weight"]
+            .gt(0)
+            .all()
+        )
+
+        forest = self.prune_unrooted(forest)
+        if not nx.is_forest(forest):
+            logger.error(
+                "Calculations in Exploded Metadata can not be represented as a forest!"
+            )
+        remaining_stepparents = set(self.stepparents(forest))
+        if remaining_stepparents:
+            logger.info(f"{remaining_stepparents=}")
+
+        # There are a few rare instances where a particular node is specified with more
+        # than one weight (-1 vs. 1) and in those cases, we always want to keep the
+        # weight of -1, since it affects the overall root->leaf calculation outcome.
+        # Maybe this should actually happen in set_forest_attributes?
+        multi_valued_weights = (
+            self.exploded_calcs.groupby(self.calc_cols, dropna=False)["weight"]
+            .transform("nunique")
+            .gt(1)
+        )
+        calcs_to_drop = multi_valued_weights & (self.exploded_calcs.weight == 1)
+        deduplicated_calcs = self.exploded_calcs.loc[~calcs_to_drop]
```

Comment:
how many true `multi_valued_weights` do we expect? it might be good to add a little assertion with some expected value to make sure we don't end up unexpectedly dropping a ton.

Reply:
Good idea. It should be very small proportion compared to the number of overall calculations.